es/es-server/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -72,8 +73,18 @@
  */
 public final class IndexModule {
 
-    public static final Setting<Boolean> NODE_STORE_ALLOW_MMAP = Setting.boolSetting(
-        "node.store.allow_mmap", true, Property.NodeScope);
+    @Deprecated
+    public static final Setting<Boolean> NODE_STORE_ALLOW_MMAPFS = Setting.boolSetting(
+        "node.store.allow_mmapfs", true, Property.NodeScope, Property.Deprecated);
+
+    public static final Setting<Boolean> NODE_STORE_ALLOW_MMAP = new Setting<>(
```

Comment:
minor nitpick: the simpler overload could be used which would use the default of the fallback setting: `new Setting<>("node.store.allow_mmap", NODE_STORE_ALLOW_MMAPFS, Booleans::parseBoolean, Property.NodeScope)`